### PR TITLE
Linkage

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Prepare many things to keep linking on Windows after the executable gets split up.

Another preliminary chore , but also a sweep touching many lines in the second commit.

Please review the first commit, and glance over the second and confirm I'm only adding AUDACITY_DLL_API in many places.

### CODE REVIEW

**Review by IGBC — APPROVED** [review]
Verified PR is clean of any code for purposes other than those described in the PR.

**Comment by IGBC** [comment]
Out of curiousity could someone explain how the AUDACITY_DLL_API decorator on structs and functions effects linking? I am new to how linking on windows works

**Review by lllucius — APPROVED** [review]
Looks fine to me as well.

**Comment by Paul-Licameli** [comment]
> Verified PR is clean of any code for purposes other than those described in the PR.

@IGBC, I am warning you once that I will treat all of your comments on my pull requests as unwelcome spam, and summarily delete them, unless withing 24 hours, I see an apology to my colleage Steve Daulton for the bullying tone you took in this comment.

https://github.com/audacity/audacity/issues/699#issuecomment-835600912

If you want to be a serious participant in the conversation about Audacity, approach the established team members with more humility and respect.

**Comment by lllucius** [comment]
> 
> 
> > Verified PR is clean of any code for purposes other than those described in the PR.
> 
> @IGBC, I am warning you once that I will treat all of your comments on my pull requests as unwelcome spam, and summarily delete them, if after 24 hours, I do not see an apology to my colleague Steve Daulton for the bullying tone you took in this comment.
> 
Oh, you didn't know Paul?  Lauren is a member of the open source S.W.A.T. team and is our new PR auditor.  She takes her commitment to Audacity VERY seriously too since she's been a (self appointed) "member" for all of a few days.  I hope she can commit the years upon years that the rest of the team has put into the project.

**Comment by Be-ing** [comment]
By all means @IGBC, keep an eye on pull requests being opened. But sarcastic code reviews and comments don't help anything (that goes for both sides).

**Comment by Paul-Licameli** [comment]
> > > Verified PR is clean of any code for purposes other than those described in the PR.
> > 
> > 
> > @IGBC, I am warning you once that I will treat all of your comments on my pull requests as unwelcome spam, and summarily delete them, if after 24 hours, I do not see an apology to my colleague Steve Daulton for the bullying tone you took in this comment.
> 
> Oh, you didn't know Paul? Lauren is a member of the open source S.W.A.T. team and is our new PR auditor. She takes her commitment to Audacity VERY seriously too since she's been a (self appointed) "member" for all of a few days. I hope she can commit the years upon years that the rest of the team has put into the project.

I share your sense of frustration.  But I will also allow those who know something about code (and this one might), and who want to offer serious suggestions (rather than presuming to dictate to us how we must do things), a chance to change their tone.  So I'm withholding my own sarcasm.

I don't want an epic argument about etiquette starting here either.  So please, enough said from us, unless @IGBC has an answer.

The pull request, meanwhile, is merged (by me) after your examination of it.

**Comment by IGBC** [comment]
> > Verified PR is clean of any code for purposes other than those described in the PR.
> 
> @IGBC, I am warning you once that I will treat all of your comments on my pull requests as unwelcome spam, and summarily delete them, if after 24 hours, I do not see an apology to my colleague Steve Daulton for the bullying tone you took in this comment.
> 
> [#699 (comment)](https://github.com/audacity/audacity/issues/699#issuecomment-835600912)
> 
> If you want to be a serious participant in the conversation about Audacity, approach the established team members with more humility and respect.
> 
> As all serious participants should treat each other.

Apologies to @SteveDaulton for the tone I took with him in  [#699 (comment)](https://github.com/audacity/audacity/issues/699#issuecomment-835600912). Full disclosure I was actually drunk when I posted that. I consider it a lesson learnt I probably shouldn't log into GitHub at all when not in a professional mindset. I wish to also extend my apologies to anyone else who felt attacked or unwelcome because of my conduct in #699

> > > Verified PR is clean of any code for purposes other than those described in the PR.
> > 
> > 
> > @IGBC, I am warning you once that I will treat all of your comments on my pull requests as unwelcome spam, and summarily delete them, if after 24 hours, I do not see an apology to my colleague Steve Daulton for the bullying tone you took in this comment.
> 
> Oh, you didn't know Paul? Lauren is a member of the open source S.W.A.T. team and is our new PR auditor. She takes her commitment to Audacity VERY seriously too since she's been a (self appointed) "member" for all of a few days. I hope she can commit the years upon years that the rest of the team has put into the project.

Well ouch. But mostly not untrue. In #835 I did point out that I would be monitoring all commits from now on. And I promise you I will not be going away in a few days either. I don't consider it some sort of open source SWAT manouvour either. I think its an important roll in the foss community.

As for my remark in my code audit. It is not intended sarcastically or with malice, only as a notice to anyone who looks at this PR and sees 1000 lines changed that they have been checked and that we don't need a repeat of #835. Nobody (especially me) wants all that drama to happen again. 

So in summary. apologies to @SteveDaulton  and @Paul-Licameli. I am here to help. Message received, I will take a more relaxed tone with the community from here on out.

**Comment by Paul-Licameli** [comment]
I accept your apology, on my own behalf.  I won't speak for Steve.

As for watching the commits, you may have noticed a big line count in this one.  True.  But as you see, it was one of those boring routine sweeps of things.  Nothing like the inclusion of a whole new library.
